x/feedist/handler.go

```diff
@@ -253,48 +241,33 @@ func distribute(db weave.KVStore, ctrl CashController, source weave.Address, rec
 			continue
 		}
 
+		// store sent and rest values for sanity check
+		zeroCoin := x.NewCoin(0, 0, c.Ticker).WithIssuer(c.Issuer)
+		totalRest, totalSent := zeroCoin, zeroCoin
+
 		// Rest of the division can be ignored, because we transfer
 		// funds to each recipients separately. Any leftover will be
 		// left on the recipients account.
-		one, _ := c.Divide(chunks)
-
 		for _, r := range recipients {
-			amount := one.Multiply(int64(r.Weight / div))
+			amount, rest := c.Multiply(int64(r.Weight)).Divide(totalWeight)
```

Comment:
I like this approach more than computing greatest common division. If we multiply fist (this is the correct order) then it is even more likely now that int64 capacity will be exceeded. To avoid it, `Coin.Multiply` should return an error as well.

Is it worth changing `Coin.Multiply` method? Should `Coin` use `big` instead of int64?

Reply:
How about a new BigCoin type for high precision math?

Reply:
Please don't overthink so much....

Note we use `int64` for Whole and Fractional.
The maximum Fractional value is 10^9, which is around 30 bits...

The maximum Whole value (total tokens) should never be more than 10^12 (even stellar and dodge are around 10^11, and the [maximum supply listed here](https://coinmarketcap.com/coins/views/market-cap-by-total-supply/) is 600*10^9.

If we assume 10^12 = 40 bits is sufficient for representing the entire token supply of any possible cryptocurrency (we can enforce that in coins if we want as a max account balance), then we can limit weights to 20 bits (10^6) and guarantee that our math will never overflow, even if we have to distribute the entire token supply of the biggest cryptocurrency.

I would avoid using bigints unless absolutely necessary.

Reply:
@ethanfrey can you explain why `BigCoin` type is not a good idea? I was about to implement it but now I am convinced that I am missing something important.